From 4aa5b95c1e3dfc70e12196145b8f88672ec40a24 Mon Sep 17 00:00:00 2001
From: Frank Wang <frank.wang@rock-chips.com>
Date: Mon, 31 Oct 2016 11:00:56 +0800
Subject: [PATCH] usb: dwc_otg_310: fix smatch warnings relate to locking
 imbalance.

This patch corrects the below locking imbalance warnings:

drivers/usb/dwc_otg_310/dwc_otg_driver.c:406 dwc_otg_force_device()
warn: inconsistent returns 'irqsave:flags'.
  Locked on:   line 390
  Unlocked on: line 406

Change-Id: Id79ebe0d41fbbab38c59384e23d7e2133c436c96
Signed-off-by: Frank Wang <frank.wang@rock-chips.com>
---
 drivers/usb/dwc_otg_310/dwc_otg_driver.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/usb/dwc_otg_310/dwc_otg_driver.c b/drivers/usb/dwc_otg_310/dwc_otg_driver.c
index eaf6772c9c18..b2d5b7f8b04a 100644
--- a/drivers/usb/dwc_otg_310/dwc_otg_driver.c
+++ b/drivers/usb/dwc_otg_310/dwc_otg_driver.c
@@ -386,6 +386,7 @@ void dwc_otg_force_device(dwc_otg_core_if_t *core_if)
 	local_irq_save(flags);
 
 	if (core_if->op_state == B_PERIPHERAL) {
+		local_irq_restore(flags);
 		printk
 		    ("dwc_otg_force_device,already in B_PERIPHERAL,everest\n");
 		return;
-- 
2.35.3

